src/backend/app/Http/Controllers/Reports.php

```diff
@@ -444,7 +437,7 @@ private function addConnectionGroupsToXLS(
 
             $meters = $connectionGroup->meters()->get();
 
-            if ($meters !== null) {
+            if (!$meters->isEmpty()) {
```

Comment:
Do we need to have a null check just to be safe here? i.e

```php
if(isset($meters) and !$meters->isEmpty()) {
}
```

Reply:
But, yeah, it's technically changing the logic. If, somehow `$meters` IS null, then modified code will fail.

Let me add your suggestions 👍

Reply:
Just tried that, but

```
Variable $meters in isset() always exists and is not nullable. 
```

🙈 

The reason here is: Earlier in the function we have:

```php
if (!isset($connectionGroup->name)) {
    continue;
}
```

The only way `$meter` could ever be `null` is when `$connectionGroup` is unset. Which is guarded by the above.